src/wallet/walletdb.cpp

```diff
@@ -885,7 +885,7 @@ DBErrors WalletBatch::LoadWallet(CWallet* pwallet)
 
     // Last client version to open this wallet, was previously the file version number
     int last_client = CLIENT_VERSION;
-    m_batch->Read(DBKeys::VERSION, last_client);
+    bool has_last_client = m_batch->Read(DBKeys::VERSION, last_client);
 
     int wallet_version = pwallet->GetVersion();
     pwallet->WalletLogPrintf("Wallet File Version = %d\n", wallet_version > 0 ? wallet_version : last_client);
```

Comment:
unrelated: Any reason for the `> 0` check? If the minimum version needed to read the wallet is `0`, it should probably be printed as-is?

Reply:
yeah, that doesn't make sense. Even `0.4.0` wallet version was `40000`.